doc/doxygen/scripts/set_canonical_doxygen.pl

```diff
@@ -0,0 +1,93 @@
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2019 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+# This script sets the canonical webpage for all the webpages of the
+# doxygen documentation. For more information see
+# https://en.wikipedia.org/wiki/Canonical_link_element
+#
+# This script is invoked by CMake. To add the canonical link to the webpages of
+# the deal.ii repository you can use the script
+# contrib/utilities/set_canonical_webpages.py
+
+use strict;
+use warnings;
+
+use File::Find qw(find);
+
+my @html;
+my $link_start = '<link rel="canonical" href="https://www.dealii.org/current/doxygen/';
+
+find(sub {
+	  if ($_ =~ /\.html$/)
+	  {
+	       push(@html, $File::Find::name);
+	  }
+     }, "./");
+chomp(@html);
+
+for my $filename (@html) {
+     if ($filename eq "./header.html")
+     {
+	  # there is no relevant content in the header
```

Comment:
I wasn't sure on what to add here - the original script uses `pass`. Is the intent to not modify `header.html`?

Reply:
Yes, but it really should have said ‘continue’. Since, if I recall correctly, there is no match, using the wrong one didn’t matter.

Reply:
Added `next` in 33c9950. Alternatively, `if ($_ =~ /\.html$/ and $_ ne "header.html")` could be used, which is shorter (but perhaps less clear).